### ISSUE TITLE
Lack of space after Sphinx role not detected

### ISSUE THREAD

**Issue by AlexWaygood** [sphinx-contrib/sphinx-lint#60]
https://github.com/python/cpython/pull/103348 initially (before it was fixed) contained this snippet, which passed sphinx-lint but failed the Docs build (see https://github.com/python/cpython/pull/103348#discussion_r1160912155):

```rst
:ref:`"What's new in Python 3.12" <whatsnew-typing-py312>`for more details.
```

This fails the Docs build because of the lack of space between the second backtick and the word "for".
